iroha/src/query.rs

```diff
@@ -40,6 +42,78 @@ impl TryFrom<SignedQueryRequest> for VerifiedQueryRequest {
     }
 }
 
+/// Unsupported version error
+#[derive(Clone, Copy, Eq, PartialEq, Debug)]
+pub struct UnsupportedVersionError {
+    /// Version that we got
+    pub version: u8,
+}
+
+impl std::error::Error for UnsupportedVersionError {}
+
+impl std::fmt::Display for UnsupportedVersionError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "Unsupported version. Expected version 1, got: {}",
+            self.version
+        )
+    }
+}
+
+/// Decode verified query error
+#[derive(Error, Debug)]
+pub enum DecodeVerifyQueryError {
```

Comment:
I would name it `AcceptQueryError` as bothe decoding and verifying is a part of accepting it at the torii endpoint.

Reply:
May be `VerifyQueryError`?

Reply:
It already includes `VerifyQuery` error, so the enum name should be more high level.